packages/aws-cdk/test/integ/cli/test-backwards-compat.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -euo pipefail
+scriptdir=$(cd $(dirname $0) && pwd)
+source ${scriptdir}/common.bash
+# ----------------------------------------------------------
+
+tmpdir=$(dirname $(mktemp -u))
+casmdir=$tmpdir/cdk-integ-cx
+stderrfile=$casmdir/cdk.err
+
+# Copy the files from a cloud assembly source dir into the
+# tempdir. Evaluate .js files found their (interpret them
+# as templates).
+function prepare_cloud_assembly() {
+    echo "ASSEMBLY ${1}"
+    rm -rf $casmdir
+    mkdir -p $casmdir
+    cp -R $1/* $casmdir
+
+    # Execute templates to produce file with the same name
+    # but without .js extension
+    shopt -s nullglob
+    for template in $casmdir/*.js; do
+        node $template > ${template%.js}
+    done
+}
+
+# Assert that there was no providerError in CDK's stderr
+# Because we rely on the app/framework to actually error in case the
+# provider fails, we inspect the logs here.
+function assert_no_error() {
+    if grep '$providerError' $stderrfile; then
+        cat $stderrfile >&2
+        echo "There was an error executing the context provider!" >&2
```

Comment:
I don't think the messaging include which test case failed; and we should include it.

Also, it's good practice to put a sample of the output when tests fail into the PR description; just to validate that the messaging looks good.